x-pack/test/functional/page_objects/infra_logs_page.ts

```diff
@@ -6,8 +6,20 @@
 
 // import testSubjSelector from '@kbn/test-subj-selector';
 // import moment from 'moment';
-
+import { encode, RisonValue } from 'rison-node';
 import { FtrProviderContext } from '../ftr_provider_context';
+import { LogPositionUrlState } from '../../../../x-pack/plugins/infra/public/containers/logs/log_position/with_log_position_url_state';
+import { FlyoutOptionsUrlState } from '../../../../x-pack/plugins/infra/public/containers/logs/log_flyout';
+
+export interface TabsParams {
+  stream: {
+    logPosition?: Partial<LogPositionUrlState>;
+    flyoutOptions?: Partial<FlyoutOptionsUrlState>;
+  };
+  settings: never;
+  'log-categories': any;
+  'log-rate': any;
+}
```

Comment:
How about splitting this up per tab and use overloads on the `navigateToTab`?

```suggestion
export interface LogStreamTabParameters {
  logPosition?: Partial<LogPositionUrlState>;
  flyoutOptions?: Partial<FlyoutOptionsUrlState>;
}

export interface LogRateTabParameters {
}

export interface LogCategoriesTabParameters {
}
```

Reply:
See https://github.com/elastic/kibana/pull/54280#discussion_r390571703